### PR TITLE
#162170676 Comment on Article

### PR DESCRIPTION
**Pull request by luqmanoop**
#### What does this PR do?
Creates a comment component with logic to allow a user to comment on an article

#### Description of Task to be completed?
Have user be able to comment on article(s)

#### How should this be manually tested?
```bash
git clone git@github.com:andela/balder-ah-frontend.git

cd balder-ah-frontend

npm install

npm run start:dev
```

#### Any background context you want to provide?
The `CommentForm` component is not currently being displayed on any route. To use it, render it in on e.g. homepage `/` like below

```js
...
import CommentForm from '../comment/CommentForm'

... // inside render

<CommentForm articleSlug="nice-article-slug" />
```

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170676](https://www.pivotaltracker.com/story/show/162170676)

#### Screenshots (If applicable)
screen shot 2019-01-24 at 2 07 20 am

#### Questions:
N/A

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -24,6 +24,7 @@
     -->
     <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" crossorigin="anonymous">
     <title>Balder Author's Haven</title>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/css/materialize.min.css">
```
Comment (akhilome) [inline public/index.html]: Materialize is already being imported on line 32.

File: src/__mocks__/helpers.jsx [hunk 2]
```diff
@@ -11,4 +15,12 @@ const renderWithRouter = (
   history,
 });
 
+export const renderWithRedux = (
```
Comment (akhilome) [inline src/__mocks__/helpers.jsx:18]: Nice!!!!!!!!!

Thanks a ton for doing this! 🙏

File: src/actions/auth.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { AUTHENTICATE_USER } from './types';
+import axios from '../utils/axiosInstance';
+import authUtils from '../utils/auth';
+
+export const getLoggedInUser = () => async (dispatch) => {
```
Comment (akhilome) [inline src/actions/auth.js:5]: 🙏

**Review by frostyblok — APPROVED** [review]

File: src/App.jsx [hunk 4]
```diff
@@ -11,33 +12,34 @@ import SocialLogin from './components/auth/SocialMediaLogin';
...
-        <Route path="/create-article" component={CreateArticle} />
-        <Route path="/articles/:slug" component={ViewArticle} />
-        <Route path="/profile" component={ProfilePage} />
-        <Route path="/social" component={SocialLogin} />
-      </Switch>
-      <Footer />
-    </Fragment>
+class App extends Component {
+  async componentDidMount() {
+    const { getLoggedInUser: getUser } = this.props;
```
Comment (oluwajuwon) [inline src/App.jsx:18]: good initiative here 👍

File: src/components/article/ViewArticle.jsx [hunk 5]
```diff
@@ -37,77 +44,92 @@ class ViewArticle extends React.Component {
...
+
+              <hr />
+              <div className="article-comments">
+                {isLoggedIn ? (
+                  <CommentForm articleSlug={slug} />
+                ) : (
+                  <p>
+                    Please
+                    {' '}
+                    <Link to="/">log in</Link>
```
Comment (oluwajuwon) [inline src/components/article/ViewArticle.jsx]: hi @codeshifu you're saying this should link back to the login page, but the link goes to the index page. It should link to `/login`.

File: src/components/comment/CommentForm.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        border-radius: 5px;
+        font-size: 16px;
+        min-height: 100px;
+        outline: transparent;
+        padding: 15px;
+        resize: none;
+        text-align: left;
+
+    }
+}
```
Comment (akhilome) [inline src/components/comment/CommentForm.scss]: Newline.

File: src/components/presentation/PreLoader.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (akhilome) [inline src/components/presentation/PreLoader.jsx:1]: Thanks for adding this in as well.

**Review by akhilome — COMMENTED** [review]
What's good @codeshifu?

I just went through your PR and I must commend the effort that went into making this output as elegant as it is.

Kudos, man.

I left a few comments below, please peruse them.

**Review by oluwajuwon — COMMENTED** [review]
Hi @codeshifu nice work 👍 .

I left some comments on your PR please go through them 😄

**Review by akhilome — APPROVED** [review]

**Review by oluwajuwon — APPROVED** [review]

File: src/components/article/CreateArticle.jsx [hunk 8]
```diff
@@ -41,26 +34,24 @@ class CreateArticle extends React.Component {
...
     }
-  }
+  };
 
   render() {
     const { message } = this.state;
+    const { isLoggedIn } = this.props;
+
+    if (isLoggedIn === null) return null;
+    if (!isLoggedIn) return <Redirect to="/login" />;
```
Comment (omokaro-faith) [inline src/components/article/CreateArticle.jsx:46]: @codeshiffu, You would want to wrap Redirect inside of `BrowserRouter`

File: src/components/auth/Signup.jsx [hunk 9]
```diff
@@ -56,6 +56,10 @@ class Signup extends Component {
     const {
       message, username, email, password, confirmPassword,
     } = this.state;
+
+    const { auth } = this.props;
+    if (auth.isLoggedIn === null) return null;
+    if (auth.isLoggedIn) return <Redirect to="/" />;
```
Comment (omokaro-faith) [inline src/components/auth/Signup.jsx:62]: Here also.

File: src/components/comment/CommentForm.scss (hunk 6 above)
Comment (omokaro-faith) [reply src/components/comment/CommentForm.scss]: You didn't modify this.

File: src/components/article/CreateArticle.jsx (hunk 8 above)
Comment (luqmanoop) [reply src/components/article/CreateArticle.jsx:46]: @omokaro-faith thanks for taking the time to review. Can you please state the reason why that's necessary?

I found [this example](https://reacttraining.com/react-router/web/example/auth-workflow) of using `<Redirect />` from the official `react-router` site.

File: src/components/article/ViewArticle.jsx (hunk 5 above)
Comment (luqmanoop) [reply src/components/article/ViewArticle.jsx]: @oluwajuwon Thanks, mate.

**Pull request merged**